accounts/abi/argument.go

```diff
@@ -196,6 +196,9 @@ func (arguments Arguments) UnpackValues(data []byte) ([]interface{}, error) {
 	virtualArgs := 0
 	for index, arg := range nonIndexedArgs {
 		marshalledValue, err := toGoType((index+virtualArgs)*32, arg.Type, data)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
should this be made as a PR to upstream?

Reply:
yep both to geth and coreth